facebook/detectors/rce/cisco_smi/src/main/java/com/google/tsunami/plugins/detectors/rce/cisco_smi/CiscoSMIDetector.java

```diff
@@ -0,0 +1,135 @@
+public final class CiscoSMIDetector implements VulnDetector {
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+
+  private final Clock utcClock;
+  private final SocketFactory socketFactory;
+
+  @Inject
+  public CiscoSMIDetector(@UtcClock Clock utcClock, SocketFactory socketFactory) {
+    this.utcClock = checkNotNull(utcClock);
+    this.socketFactory = checkNotNull(socketFactory);
+  }
+
+  @Override
+  public DetectionReportList detect(
+      TargetInfo targetInfo, ImmutableList<NetworkService> matchedServices) {
+    return DetectionReportList.newBuilder()
+        .addAllDetectionReports(
+            matchedServices.stream()
+                .filter(this::isServiceVulnerable)
+                .map(networkService -> buildDetectionReport(targetInfo, networkService))
+                .collect(toImmutableList()))
+        .build();
+  }
+
+  private boolean isServiceVulnerable(NetworkService networkService) {
+    byte[] actualResponse = new byte[24];
+    // See https://github.com/Cisco-Talos/smi_check for more info about the payload
+    byte[] request = {
+        0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x01,
+        0x00, 0x00, 0x00, 0x04, 0x00, 0x00, 0x00, 0x08,
+        0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x00,
+    };
+    byte[] expectedResponse = {
+        0x00, 0x00, 0x00, 0x04, 0x00, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x03, 0x00, 0x00, 0x00, 0x08,
+        0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x00,
+    };
+    HostAndPort hp = NetworkEndpointUtils.toHostAndPort(networkService.getNetworkEndpoint());
+
+    try {
+        Socket socket = socketFactory.createSocket(hp.getHost(), hp.getPort());
+        socket.setSoTimeout(2000);
+        socket.getOutputStream().write(request);
+        socket.getInputStream().read(actualResponse, 0, actualResponse.length);
+        socket.close();
+    } catch (Exception e) {
+        logger.atWarning().withCause(e).log("Unable to communicate with '%s'.", hp.toString());
+        return false;
+    }
+
+    if (Arrays.equals(actualResponse, expectedResponse)) {
+        return true;
+    } else if (actualResponse.length < 1) {
+        logger.atInfo().log("SMI is enabled but not vulnerable");
+    }
+    return false;
+  }
+
+  private DetectionReport buildDetectionReport(
+      TargetInfo scannedTarget, NetworkService vulnerableNetworkService) {
+    return DetectionReport.newBuilder()
+        .setTargetInfo(scannedTarget)
+        .setNetworkService(vulnerableNetworkService)
+        .setDetectionTimestamp(Timestamps.fromMillis(Instant.now(utcClock).toEpochMilli()))
+        .setDetectionStatus(DetectionStatus.VULNERABILITY_VERIFIED)
+        .setVulnerability(
+            Vulnerability.newBuilder()
+                .setMainId(
+                    VulnerabilityId.newBuilder()
+                        .setPublisher("Cisco")
+                        .setValue("cisco-sa-20170214-smi"))
```

Comment:
Do you mind updating this to all uppercase letters, with words separated by underscores, i.e. `CISCO_SA_20170214_SMI` to make this consistent within this repository?